DumpRenderTree does not seem to support PingLoad to HTTPS

brentfulgham opened the pull request:
#### 6db9bb4cc51f9fd26437c89e48a570944989bcfb
<pre>
DumpRenderTree does not seem to support PingLoad to HTTPS
<a href="https://bugs.webkit.org/show_bug.cgi?id=246719">https://bugs.webkit.org/show_bug.cgi?id=246719</a>
&lt;rdar://101040717&gt;

Reviewed by Chris Dumez.

Revise DumpRenderTree to handle ServerTrust protection space evaluations,
and to allow them under the same cases as Modern WebKit.

* LayoutTests/platform/mac-wk1/TestExpectations: Correct some expectations.
* Source/WebCore/page/DeprecatedGlobalSettings.h:
* Source/WebCore/testing/js/WebCoreTestSupport.cpp:
(WebCoreTestSupport::allowsAnySSLCertificate): Added getter to expose state to DRT.
* Source/WebCore/testing/js/WebCoreTestSupport.h:
* Tools/DumpRenderTree/TestRunner.cpp:
(TestRunner::allowsAnySSLCertificate): Added.
* Tools/DumpRenderTree/TestRunner.h:
(TestRunner::allowAnyHTTPSCertificateForAllowedHosts const): Added.
(TestRunner::setAllowAnyHTTPSCertificateForAllowedHosts): Added.
(TestRunner::localhostAliases const): Added.
(TestRunner::setLocalhostAliases): Added.
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(initializeGlobalsFromCommandLineOptions): Update to recognize the allowed hosts test runner argument.
(dumpRenderTree):
(runTest):
* Tools/DumpRenderTree/mac/ResourceLoadDelegate.mm:
(canAuthenticateServerTrustAgainstProtectionSpace): Added.
(-[ResourceLoadDelegate webView:resource:didReceiveAuthenticationChallenge:fromDataSource:]): Updated to behave
the same way as WebKitTestRunner.
(-[ResourceLoadDelegate webView:resource:canAuthenticateAgainstProtectionSpace:forDataSource:]): Added.

Canonical link: <a href="https://commits.webkit.org/256173@main">https://commits.webkit.org/256173@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca607b87019691dc9b156f3a652412304c51f9a7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94921 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4069 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27825 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104527 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164791 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98916 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4154 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/32250 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87224 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100448 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100590 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3001 "Passed tests") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/81501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29987 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84915 "Found 1 new API test failure: TestWebKitAPI.TLSVersion.LegacySubresources (failure)") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72879 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38662 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/18297 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36494 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19578 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4254 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40419 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/81501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42395 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38810 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM assuming the bots are happy.

aproskuryakov commented on Tools/DumpRenderTree/mac/DumpRenderTree.mm:
```diff
@@ -1184,6 +1190,9 @@ void dumpRenderTree(int argc, const char *argv[])
 
     [NSURLRequest setAllowsAnyHTTPSCertificate:YES forHost:@"localhost"];
     [NSURLRequest setAllowsAnyHTTPSCertificate:YES forHost:@"127.0.0.1"];
```
Are these still necessary, given that we handle the delegate call now?

brentfulgham committed:
DumpRenderTree does not seem to support PingLoad to HTTPS
https://bugs.webkit.org/show_bug.cgi?id=246719
<rdar://101040717>

Reviewed by Chris Dumez.

Revise DumpRenderTree to handle ServerTrust protection space evaluations,
and to allow them under the same cases as Modern WebKit.

* LayoutTests/platform/mac-wk1/TestExpectations: Correct some expectations.
* Source/WebCore/page/DeprecatedGlobalSettings.h:
* Source/WebCore/testing/js/WebCoreTestSupport.cpp:
(WebCoreTestSupport::allowsAnySSLCertificate): Added getter to expose state to DRT.
* Source/WebCore/testing/js/WebCoreTestSupport.h:
* Tools/DumpRenderTree/TestRunner.cpp:
(TestRunner::allowsAnySSLCertificate): Added.
* Tools/DumpRenderTree/TestRunner.h:
(TestRunner::allowAnyHTTPSCertificateForAllowedHosts const): Added.
(TestRunner::setAllowAnyHTTPSCertificateForAllowedHosts): Added.
(TestRunner::localhostAliases const): Added.
(TestRunner::setLocalhostAliases): Added.
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(initializeGlobalsFromCommandLineOptions): Update to recognize the allowed hosts test runner argument.
(dumpRenderTree):
(runTest):
* Tools/DumpRenderTree/mac/ResourceLoadDelegate.mm:
(canAuthenticateServerTrustAgainstProtectionSpace): Added.
(-[ResourceLoadDelegate webView:resource:didReceiveAuthenticationChallenge:fromDataSource:]): Updated to behave
the same way as WebKitTestRunner.
(-[ResourceLoadDelegate webView:resource:canAuthenticateAgainstProtectionSpace:forDataSource:]): Added.

Canonical link: https://commits.webkit.org/256173@main
